modules/product/product.service.js

```diff
@@ -34,43 +32,50 @@ class ProductsService {
         },
       };
     }
-    if(price){
+    if (price.length) {
       filter.basePrice = {
-          $gte: price[0], 
-          $lte: price[1]
+        $gte: price[0],
+        $lte: price[1],
       };
     }
     return filter;
-  };
+  }
 
-  async getProductsByOptions({filter, skip, limit, sort, search}) {
-    
+  getProducts({
+    filter, skip, limit, sort, search,
+  }) {
     const isNotBlank = str => !(!str || str.trim().length === 0);
     const filters = this.filterItems(filter);
 
     if (isNotBlank(search)) {
-      filters.$or =  [
-        { name: { $elemMatch: { value: { $regex: new RegExp(search,'i') } } } },
-        { description: { $elemMatch: { value: { $regex: new RegExp(search,'i') } } } },
+      filters.$or = [
+        {
+          name: { $elemMatch: { value: { $regex: new RegExp(search, 'i') } } },
+        },
+        {
+          description: {
+            $elemMatch: { value: { $regex: new RegExp(search, 'i') } },
+          },
+        },
       ];
     }
-    
-    return await Products.find( filters)
+
+    return Products.find(filters)
       .skip(skip)
       .limit(limit)
       .sort(sort);
   }
 
-  updateProductById(id, products) {
+  updateProduct(id, products) {
     return Products.findByIdAndUpdate(id, products);
```

Comment:
Please, add `{ new: true }` here to return newly updated data

Reply:
done